frontend/src/metabase/visualizations/visualizations/WaterfallChart.jsx

```diff
@@ -53,6 +55,21 @@ export default class WaterfallChart extends LineAreaBarChart {
 
   static renderer = waterfallRenderer;
 
+  static checkRenderable = (series, settings) => {
+    super.checkRenderable(series, settings);
+
+    const [{ card }] = series;
+    const question = new Question(card);
+    // query().breakouts() does not exist in native queries
+    const queryBreakouts = question.query().breakouts?.() || [];
+
+    if (queryBreakouts.length > this.maxDimensionsSupported) {
+      throw new Error(
+        t`${this.uiName} chart does not support queries with multiple groupings`,
```

Comment:
@npfitz I'd go with "does not support multiple groupings" for the copy here, otherwise LGTM